public/app/plugins/panel/heatmap/utils.ts

```diff
@@ -782,10 +789,7 @@ export function heatmapPathsPoints(opts: PointsBuilderOpts, exemplarColor: strin
           let isSparseHeatmap = scaleY.distr === 3 && scaleY.log === 2;
 
           if (!isSparseHeatmap) {
-            yVal -= 0.5; // center vertically in bucket (when tiles are le)
-            // y-randomize vertically to distribute exemplars in same bucket at same time
-            let randSign = Math.round(Math.random()) * 2 - 1;
-            yVal += randSign * 0.5 * Math.random();
```

Comment:
this removes exemplar randomization within each cell and provides stability, but also maybe overlap. probably a better trade-off. we can look at clustering or multi-hover in the future.